### PR TITLE
Rewrite and update of `dvc config` command ref.

### PR DESCRIPTION
**Pull request by jorgeorpinel**

### CODE REVIEW

File: src/DownloadButton/index.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import styled, { css } from 'styled-components'
 import isClient from '../utils/isClient'
 import { logEvent } from '../utils/ga'
 
-const VERSION = `0.40.1`
+const VERSION = `0.40.2`
```
Comment (shcheklein) [inline src/DownloadButton/index.js]: merge or rebase on top of master?

File: static/docs/commands-reference/config.md [hunk 2]
```diff
@@ -1,38 +1,42 @@
...
 ```usage
-    usage: dvc config [-h] [-q | -v] [--global] [--system] [--local] [-u] name
-                      [value]
+    usage: dvc config [-h] [-q | -v] [--global] [--system] [--local]
+                      [-u] name [value]
 
     positional arguments:
-        name                  Option name
-        value                 Option value
+        name                  Setting name
```
Comment (shcheklein) [inline static/docs/commands-reference/config.md]: we should make a PR to DVC to change these descriptions

File: static/docs/commands-reference/config.md [hunk 3]
```diff
@@ -41,156 +45,163 @@ share through Git.
...
+sections in the project config file (`.dvc/config`).
+
+### remote
+
+These are sections in the config file that describe particular remotes. See `dvc
+remote` for more info.
+
+### core
+
+This is the main section with the general settings:
```
Comment (shcheklein) [inline static/docs/commands-reference/config.md]: let's double check. I'm pretty sure that we now have a whole bunch of new options.

File: static/docs/commands-reference/config.md [hunk 4]
```diff
@@ -41,156 +45,163 @@ share through Git.
...
+
+* `cache.dir` - set/unset cache directory location. A correct value must be either an
+  absolute path or a path **relative to the config file location**. The default
+  value is `cache`, which resolved relative to the default project config
+  location results in `.dvc/cache`.
+  > See also helper command `dvc cache dir` that properly transform paths
+  relative to the present working directory into relative to the project config
+  file.
+
+* `cache.protected` - makes files in the workspace read-only. Possible values
```
Comment (shcheklein) [inline static/docs/commands-reference/config.md]: should mention that if `hardlink` or `symlink` cache type is used it's highly recommend to enable this mode.

File: static/docs/commands-reference/config.md (hunk 2 above)
Comment (jorgeorpinel) [reply static/docs/commands-reference/config.md]: I did (: https://github.com/iterative/dvc/pull/1999

File: src/DownloadButton/index.js (hunk 1 above)
Comment (jorgeorpinel) [reply src/DownloadButton/index.js]: Yeah Idk what happened there. Looks like I accidentally cherry picked a commit?? Weird. I removed it now (had to force-push to my fork).

File: static/docs/commands-reference/config.md (hunk 2 above)
Comment (efiop) [reply static/docs/commands-reference/config.md]: Left a comment in iterative/dvc#1999

Comment (jorgeorpinel) [reply static/docs/commands-reference/config.md]: Thanks Ruslan. From that comment I get the sense that we don't want to change this terminology in the command output, so I can figure out some other way to make a clear distinction between the command options (`--unset`, etc) and the DVC options (`core.remote`, etc). Please confirm 😃 (and feel fee to close https://github.com/iterative/dvc/pull/1999)

File: static/docs/commands-reference/config.md (hunk 3 above)
Comment (jorgeorpinel) [reply static/docs/commands-reference/config.md]: There's tons of config sections listed in https://github.com/iterative/dvc/blob/7ceaf88c54c90d53e64c1af2fa78b5b9bd4667a0/dvc/config.py#L139-L274 but I don't know what they do. I'll ask on Discord.

Comment (jorgeorpinel) [reply static/docs/commands-reference/config.md]: Added all `core` and `cache` options but still not finished with all the other sections. Will lky

File: static/docs/commands-reference/config.md (hunk 4 above)
Comment (jorgeorpinel) [reply static/docs/commands-reference/config.md]: Added

**Comment by jorgeorpinel** [comment]
The idea of splitting up the doc definitely makes sense although it's not like `dvc remote` in that there's no sub-commands, just config options sections. So I'm not sure, you can navigate pretty easily with the right-hand side CONTENTS links I think. But lmk @shcheklein and I'll create an issue to split it up after this round is done (so I can get this one done and focus on other priorities for now). Thanks

**Comment by jorgeorpinel** [comment]
@shcheklein I think this PR is ready for merge, please review.

I also opened a new issue regarding splitting up this cmd ref: https://github.com/iterative/dvc.org/issues/340

File: static/docs/commands-reference/checkout.md [hunk 5]
```diff
@@ -15,37 +15,43 @@ Update data files and directories in workspace based on current DVC files.
...
-Note, this command does NOT copy any files (exception: `cache.type == copy`).
-Instead, DVC uses links to perform data file restoration. This is crucial for
-large files where checking out a 50Gb file might take a few minutes. With DVC
-links, restoring a 50Gb data file will take less than a second.
+Note, this command does NOT copy any files (except when the `cache.type` config
+option is `copy`). Instead, DVC uses links to perform data file restoration.
+This is crucial for large files where checking out a 50Gb file might take a few
+minutes. With DVC links, restoring a 50Gb data file will take less than a
+second.
+> Note that when linking files takes longer than expected and `cache.type` is
```
Comment (shcheklein) [inline static/docs/commands-reference/checkout.md]: add a line between paragraphs for readability?

File: static/docs/commands-reference/config.md [hunk 6]
```diff
@@ -41,155 +46,205 @@ returns current value of the option.
...
+  the cached version of a file by editing or overwriting it. Turning this config
+  option on forces you to run `dvc unprotect` before updating a file, providing
+  an additional layer of security to your data.  
+  It's highly recommended to enable this mod when `cache.type` is set to
+  `hardlink` or `symlink`. 
 
-```dvc
-    $ dvc config cache.protected true
-```
+- `cache.type` - link type that dvc should use to link data files from cache to
```
Comment (shcheklein) [inline static/docs/commands-reference/config.md]: dvc -> DVC?

**Pull request merged**